### PR TITLE
Badge组件添加点击事件&button组件click阻止冒泡

### PR DESCRIPTION
**Pull request by xiaofeng-bm**
- [ ] Component Migration

Fix #

### CODE REVIEW

**Comment by JeremyWuuuuu** [comment]
Reason for this?

**Comment by xiaofeng-bm** [comment]
> Reason for this?

When I encounter a business that clicks on the Badge component to show some details, I personally think I can add a click event to make it more flexible. Of course, I can turn off the isseue if you don't think it's necessary.

**Comment by JeremyWuuuuu** [comment]
Well, I do not think this is necessary anyway.
If you want to catch the click event, you should wrap `<el-tag>` under some other tag like `div`, then you attach `@click.stop` upon it.

File: packages/badge/src/index.vue [hunk 1]
```diff
@@ -12,6 +12,7 @@
             'is-dot': isDot
           }
         ]"
+        @click.stop="handleClick"
```
Comment (JeremyWuuuuu) [inline packages/badge/src/index.vue:15]: BTW, you can just change the code here as
```suggestion
@click.stop="$emit('click')"
```

**Comment by xiaofeng-bm** [comment]
> Well, I do not think this is necessary anyway.
> If you want to catch the click event, you should wrap `<el-tag>` under some other tag like `div`, then you attach `@click.stop` upon it.
> @HerringtonDarkholme @zazzaz any suggestions?
Thanks, you're right, I didn't think of it that way.

**Pull request closed without merging**